doc - add content for new greenplum_fdw module

lisakowen opened the pull request:
add some basic module docs for greenplum_fdw.  this is based on postgres_fdw.  the org of this page differs from the org of postgres_fdw docs.  in some cases i xref'd to the postgres docs.

doc review site link (behind vpn):  https://docs-lisa-gpfdw.sc2-04-pcf1-apps.oc.vmware.com/6-19/ref_guide/modules/greenplum_fdw.html

huiliang-liu commented on gpdb-doc/dita/ref_guide/modules/greenplum_fdw.xml:
```diff
@@ -0,0 +1,360 @@
+            <codeblock>mpp_execute 'all segments'</codeblock>
+          <p>and</p>
+            <codeblock>num_segments '<varname>num</varname>'</codeblock>
+          <p>in the <codeph>OPTIONS</codeph> clause when you create the server. Set
```
`num_segments`''s default value will be the segment number of local greenplum when not specified.

huiliang-liu commented on gpdb-doc/dita/ref_guide/modules/greenplum_fdw.xml:
```diff
@@ -0,0 +1,360 @@
+          <p>You <codeph>SELECT</codeph> from a foreign table to access the data
+            stored in the underlying remote Greenplum Database table. By default, you can
+            also modify the remote table using <codeph>INSERT</codeph>,
+            <codeph>UPDATE</codeph>, or 
```
greenplum fdw does't support update and delete, if the `mpp_execute` option is set as `all segments`.

huiliang-liu commented on gpdb-doc/dita/ref_guide/modules/greenplum_fdw.xml:
```diff
@@ -0,0 +1,360 @@
+        <dlentry>
+        <dt><codeph>updatable</codeph></dt>
+        <dd>Controls whether <codeph>greenplum_fdw</codeph> allows foreign tables
+          to be modified using <codeph>INSERT</codeph>, <codeph>UPDATE</codeph>, and
```
same as above

lisakowen commented:
thank you for reviewing, @huiliang-liu .  i incorporated your requested edits.  i removed update/delete from supported update operations, and added a related limitation.

lisakowen commented:
@adam8157 - are you interested in reviewing the greenplum_fdw docs?

adam8157 commented on gpdb-doc/dita/ref_guide/modules/greenplum_fdw.xml:
```diff
@@ -0,0 +1,362 @@
+    <topic id="topic_update">
+      <title>About the Updatability Option</title>
+      <body>
+        <p>By default, all foreign tables created with <codeph>greenplum__fdw</codeph> are
```
redundant underscore here

adam8157 reviewed (commented):
Others look good to me, and Huiliang knows greenplum_fdw much better than me. 😊
